storage/src/meta/batch.rs

```diff
@@ -0,0 +1,144 @@
+    pub fn as_slice(&self) -> &[u8] {
+        unsafe {
+            slice::from_raw_parts(
+                self as *const BatchInflateContext as *const u8,
+                size_of::<BatchInflateContext>(),
+            )
+        }
+    }
+}
+
+/// Struct to generate [BatchInflateContext] objects for batch chunks.
+pub struct BatchContextGenerator {
+    /// Buffering the to be dumped chunk data for Chunk Merging.
+    chunk_data_buf: Vec<u8>,
+    /// Storing all `BatchInflateContext` of current blob.
+    contexts: Vec<BatchInflateContext>,
+}
+
+impl BatchContextGenerator {
+    /// Get the buffer of to be dumped chunk data for batch chunk.
+    pub fn chunk_data_buf(&self) -> &Vec<u8> {
+        &self.chunk_data_buf
+    }
+
+    /// Check whether the chunk data buffer is empty.
+    pub fn chunk_data_buf_is_empty(&self) -> bool {
+        self.chunk_data_buf.is_empty()
+    }
+
+    /// Get the lenth of chunk data buffer.
+    pub fn chunk_data_buf_len(&self) -> usize {
+        self.chunk_data_buf.len()
+    }
+
+    /// Append new chunk data to the chunk data buffer.
+    pub fn append_chunk_data_buf(&mut self, chunk_data: &mut Vec<u8>) {
+        self.chunk_data_buf.append(chunk_data);
+    }
+
+    /// Clear the chunk data buffer.
+    pub fn clear_chunk_data_buf(&mut self) {
+        self.chunk_data_buf.clear();
+    }
+
+    /// Add a batch context for a dumped batch chunk.
+    pub fn add_context(&mut self, compressed_offset: u64, compressed_size: u32) {
+        let ctx = BatchInflateContext {
+            compressed_offset: u64::to_le(compressed_offset),
+            compressed_size: u32::to_le(compressed_size),
+            __reserved1: u32::to_le(0),
+            __reserved2: u64::to_le(0),
+            __reserved3: u64::to_le(0),
+            __reserved4: u64::to_le(0),
+        };
+        self.contexts.push(ctx);
+    }
+
+    /// Create a new instance of [BatchInflateContext].
+    pub fn new() -> Result<Self> {
+        Ok(Self {
+            chunk_data_buf: Vec::with_capacity(RAFS_DEFAULT_CHUNK_SIZE as usize),
+            contexts: Vec::with_capacity(10240),
+        })
+    }
+
+    /// Generate and return a v2 chunk info struct.
+    pub fn generate_chunk_info(
+        &mut self,
+        uncompressed_offset: u64,
+        uncompressed_size: u32,
+        uncompressed_batch_offset: u32,
+    ) -> Result<BlobChunkInfoV2Ondisk> {
+        let mut chunk = BlobChunkInfoV2Ondisk::default();
+        chunk.set_compressed_offset(0);
+        chunk.set_compressed_size(0);
+        chunk.set_uncompressed_offset(uncompressed_offset);
+        chunk.set_uncompressed_size(uncompressed_size);
+        chunk.set_batch(true);
+        chunk.set_batch_index(self.contexts.len() as u32);
+        chunk.set_uncompressed_batch_offset(uncompressed_batch_offset);
+        chunk.set_compressed(true); // TODO: set compressed on demand
+
+        Ok(chunk)
+    }
+
+    /// Save the batch chunk information to a file.
```

Comment:
Please help to update the comment, it's not save data to a file.

Reply:
Updated.